modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/nextcloud_strategies.rb

```diff
@@ -37,10 +37,24 @@ module NextcloudStrategies
             ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::BasicAuth.strategy
           end
 
-          UserBound = ->(user:) do
-            ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::OAuthUserToken
-              .strategy
-              .with_user(user)
+          UserBound = ->(user:, storage:) do
+            # README: This check does not match for user that are only hijacked by an IDP based login.
+            token_exchange_preferred = storage.audience.present? && user.authentication_provider.present?
```

Comment:
I'd call this `sso_preferred`, since the underlying strategy does not solely work with Token Exchange, but with all kinds of tokens obtained through SSO.

Reply:
yep, naming is hard :D